app/DAO/BakeryDatabase.scala

```diff
@@ -89,60 +90,76 @@ class BakeryDatabase @Inject() (
       .unsafeToFuture()
   }
 
-  def createProduct(newProduct: JsValue): Int = {
+  def createProduct(newProduct: productUpdateRequest): Int = {
 
     /** Steps for creating a record
       *  1. bring in the json, everything should be provided except for the uuid
       *  2. use query.update to commit the record
       *  3. run the transaction
       *  4. return a variable the controller can check to know the transaction succeeded (201 Created)
       */
-    val newName = (newProduct \ "name").asOpt[String]
-    val newQty = (newProduct \ "quantity").asOpt[Int]
-    val newPrice = (newProduct \ "price").asOpt[Double]
+    val newName = newProduct.name
+    val newQty = newProduct.quantity
+    val newPrice = newProduct.price
```

Comment:
not super important, but for the record, these three variables are superfluous

Reply:
Wrote them into the match clause instead